libcob/fileio.c

```diff
@@ -690,15 +690,16 @@ static unsigned int	bdb_lock_id = 0;
 	key.data = fld->data;			\
 	key.size = (cob_dbtsize_t) fld->size
 
-#if (DB_VERSION_MAJOR > 4) || ((DB_VERSION_MAJOR == 4) && (DB_VERSION_MINOR > 0))
```

Comment:
change to

```c
#if DB_VERSION_MAJOR >= 6 /* while the fields are part of DBT since 4.0, it wasn't
                             made part of the public API before 6.0 (DB_VERSION_FAMILY 12);
                             older versions overwrite that, see bug NNN */
```

Reply:
This does not work, even using the version 6 the test fails.

Reply:
I've also tested with version 18, available on Oracle website, it also failed

Reply:
That's interesting-  in this case you found a BDB bug.

@emilienlemaire Please change the old `#if` into a `#if 0` explaining the issue in 3-5 lines (possibly keeping the DB6 public note), as this ensures we don't wonder about that in the future and search other solutions if this change bring not-yet-know other issues.